### PR TITLE
Show console logs in case of shutdown/bootup problems

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by coolo** [comment]
this is run behind every test that is not x11test and not consoletest - and do you know how the extra esc impacts e.g. yast tests? 

-> I require at least 5 different test scenarios with this change

**Comment by okurz** [comment]
wait, should have been a post_fail_hook. Can I just define one in this class or do you have better recommendations?

**Comment by coolo** [comment]
same applies there. if the test already has a post_fail_hook yours will not be called - or you have to add SUPER calls in all of them. So I would rather add the esc explicitly in tests that boot or shutdown

**Comment by okurz** [comment]
Updated:
I moved `wait_boot` into the base class `opensusetest`. It fits there much better than other methods that are more specific. Along with `wait_boot` I provided a hook helper which I call in post_fail_hooks in nearly all cases where `wait_boot` is called.

Please recheck.

Local verification runs:
* http://lord.arch/tests/5433
* http://lord.arch/tests/5443

**Comment by okurz** [comment]
updated as discussed with @foursixnine:
* moved `post_fail_hook` into baseclass `opensusebasetest` so that it is defined for every module deriving from it. If there is a custom hook in a module and there is a use of `wait_boot` parent post fail hook is called explicitly in the module custom hook

**Comment by foursixnine** [comment]
@coolo What do you think about the last changes?... After this is merged, I'll be able to include memory dumping and improving log collection if possible.

File: tests/kernel/virtio_console.pm [hunk 1]
```diff
@@ -24,7 +24,7 @@ FIN.
 
 sub run {
     my $self = shift;
```
Comment (asmorodskyi) [inline tests/kernel/virtio_console.pm]: need to replace with my $self = @_;

File: tests/kernel/boot_ltp.pm [hunk 2]
```diff
@@ -14,11 +14,10 @@ use 5.018;
 use warnings;
 use base 'opensusebasetest';
 use testapi;
-use utils;
 
 sub run {
     my $self = shift;
```
Comment (asmorodskyi) [inline tests/kernel/boot_ltp.pm]: need to replace with my $self = @_;

File: tests/hpc/install.pm [hunk 3]
```diff
@@ -18,7 +18,6 @@
 use base "opensusebasetest";
 use strict;
 use testapi;
-use utils qw/wait_boot/;
 
 sub run() {
     my $self = shift;
```
Comment (asmorodskyi) [inline tests/hpc/install.pm]: need to replace with my $self = @_;

Comment (okurz) [inline tests/hpc/install.pm]: yes, I will do that consistently in another commit changing every appearance after another PR ensuring a consistent code style for all code.

**Pull request merged**